### PR TITLE
refactor: add externalRenderer

### PR DESCRIPTION
**Pull request by WoodNeck**
## Details
- This fixes internal typings by adding a new option to Flicking - `externalRenderer`
- This deprecates option `renderExternal`
- This fixes Vue & Vue3 Flicking's type

This is marked as `refactor` because both options are internal-only

### CODE REVIEW

**Review by mixed — APPROVED** [review]
LGTM

File: packages/vue3-flicking/src/Flicking.ts [hunk 1]
```diff
@@ -75,23 +126,19 @@ class Flicking extends Vue.with(FlickingProps) {
     if (this.status) {
       flicking.setStatus(this.status);
     }
-  }
-
-  public beforeDestroy() {
+  },
+  beforeDestroy() {
```
Comment (daybrush) [inline packages/vue3-flicking/src/Flicking.ts:130]: I think you can use beforeUnmount.

Comment (WoodNeck) [reply packages/vue3-flicking/src/Flicking.ts:130]: Looks like I should, thanks for checking 👍

**Review by daybrush — APPROVED** [review]
LGTM

**Review by ahn3407 — APPROVED** [review]
LGTM

**Pull request merged**